packages/config/src/options.js

```diff
@@ -452,10 +452,12 @@ export function getNuxtConfig (_options) {
 
   // App config (internal for nuxt2 at this stage)
   const useCDN = isUrl(options.build.publicPath) && !options.dev
+  const isRelativePublicPath = isRelative(options.build.publicPath)
+
   options.app = defu(options.app, {
     basePath: options.router.base,
-    assetsPath: useCDN ? '/' : joinURL(options.router.base, options.build.publicPath),
-    cdnURL: useCDN ? options.build.publicPath : null
+    assetsPath: useCDN || isRelativePublicPath ? '/' : joinURL(options.router.base, options.build.publicPath),
+    cdnURL: useCDN || isRelativePublicPath ? options.build.publicPath : null
```

Comment:
This is logically wrong. relative url is not cdn...

Reply:
good point. to implement this though with `assetsPath` we would have to change some logic to avoid urlJoining the router base to relative path - happy with that?

Reply:
ah - no, I'm wrong - I forgot https://github.com/nuxt-contrib/ufo/pull/15 hadn't yet been merged...

Reply:
support for falsy base added in ufo@0.6.10